### ISSUE TITLE
Preserve on reload

### ISSUE THREAD

**Issue by maximilianweidenauer** [sibvisions/reactUI#110]
It should be possible to preserve the session on reload, so that not everytime someone refreshes, the startup request gets sent again. Maybe as a flag?

**Comment by gimmixAT** [comment]
Which api endpoint should call instead of startup? Just an openscreen call for the home screen? 
For that to work we will probably have to manually cache the initial startup responses in a sessionstorage and "replay" them when the user refreshes the browser window.

**Comment by rjahn** [comment]
We don't have a special endpoint, but api/changes would work. It sends only changed components - if any.

**Comment by gimmixAT** [comment]
Opening the screen given in the url should work now. So reloading the page now "stays" on the same page and doesn't call another startup.

The only issue right now might be that I get the componentId for the openScreen call from the menu that was sent in the initial startup call. Also I have to guess which componentId to use from the navigationName I get from the url. I do this now by checking if the navigatioName is part of the componentId because the navigatioName seems to basically the screen's className sans the "WorkScreen" suffix. 

TLDR: An additional navigatioName field in the menu items sent with startup might help but are not necessary - for the demo / features app at least.

**Comment by maximilianweidenauer** [comment]
Currently there is a problem, that when you logout from let's say admin and login with features, the startup hash logs you in as admin, but the current user on the server is feature... This maybe also gets fixed when the server implements the reload command so lets see.

**Comment by rjahn** [comment]
startup hash?

It should work as follows: The clientId is the only information, required on the client. After browser reload -> send api/uiRefresh request. The response will contain the same information as returned by a startup... 

Currently, the server isn't ready, but that's the plan!

**Comment by maximilianweidenauer** [comment]
i meant startup cache, thats our temporary solution. We will wait for the server.

**Comment by rjahn** [comment]
UI refresh command is available in current server repo:

**`api/uiRefresh`**

request only requires: 

 **`clientId`**

Please do some tests because it's possible that some mechanism' are not yet working on server-side.

**Comment by maximilianweidenauer** [comment]
When I'm using the uiRefresh tables don't receive "columnLabels" and "columnNames" properties. On screens without tables it seems to work fine, but I haven't tried it with every screen

**Comment by rjahn** [comment]
Should work with latest version

**Comment by maximilianweidenauer** [comment]
Yeah it works now as expected. I'm closing this issue now, if something is still wrong we can reopen or create a new ticket.
